File: 3884-first-matching-character-from-both-ends/3884-first-matching-character-from-both-ends.js
/**
 * @param {string} s
 * @return {number}
 */
var firstMatchingIndex = function(s) {
    
    let start = 0 ;
    let end = s.length - 1 ;

    while( start <= end && s[start] != s[end] ){
        start ++ ;
        end -- ;
    }

    if( start <= end ) return start ;

    return -1 ;
};

File: 3884-first-matching-character-from-both-ends/3884-first-matching-character-from-both-ends.py
class Solution(object):
    def firstMatchingIndex(self, s):
        """
        :type s: str
        :rtype: int
        """

        start = 0 
        end = len(s) - 1 

        while( start <= end and s[start] != s[end] ):
            start += 1 
            end -= 1 
        
        if( start <= end ):
            return start 

        return -1    